game.py

```diff
@@ -85,16 +132,30 @@ def check_for_collisions(player, obstacles):
     if player.y + player.height > WINDOW_HEIGHT:
         player.y = WINDOW_HEIGHT - player.height
 
-    # This needs to happen before drawing all the other objects
-    window.fill((255, 255, 255))
+    # Calculate current frame based on animation speed
+    current_time = pygame.time.get_ticks()
```

Comment:
I would move this closer to where the current_frame is actually being used or maybe we should extract it into a different method inside the Player class?